src/Features/Core/Portable/NavigateTo/AbstractNavigateToSearchService.CachedDocumentSearch.cs

```diff
@@ -20,38 +20,20 @@
 
 namespace Microsoft.CodeAnalysis.NavigateTo;
 
-using CachedIndexMap = ConcurrentDictionary<(IChecksummedPersistentStorageService service, DocumentKey documentKey, StringTable stringTable), AsyncLazy<TopLevelSyntaxTreeIndex?>>;
-
 internal abstract partial class AbstractNavigateToSearchService
 {
     /// <summary>
-    /// Cached map from document key to the (potentially stale) syntax tree index for it we use prior to the 
-    /// full solution becoming available.  Once the full solution is available, this will be dropped
-    /// (set to <see langword="null"/>) to release all cached data.
-    /// </summary>
-    private static CachedIndexMap? s_cachedIndexMap = [];
-
-    /// <summary>
-    /// String table we use to dedupe common values while deserializing <see cref="SyntaxTreeIndex"/>s.  Once the 
-    /// full solution is available, this will be dropped (set to <see langword="null"/>) to release all cached data.
+    /// Index of data used when we're doing our search of cached data.  This is used up to the point that OOP is fully
+    /// loaded and we can transition to doing normal searches on up-to-date data.  The internal data for this is
+    /// initialized the first time we do a cached search, and the entire object itself is dropped when we're done with
+    /// it.
     /// </summary>
-    private static StringTable? s_stringTable = new();
+    private static CachedIndexMap? _cachedIndexMap_DoNotAccessDirectly = new();
 
     private static void ClearCachedData()
     {
-        // Volatiles are technically not necessary due to automatic fencing of reference-type writes.  However,
-        // i prefer the explicitness here as we are reading and writing these fields from different threads.
-        Volatile.Write(ref s_cachedIndexMap, null);
-        Volatile.Write(ref s_stringTable, null);
-    }
-
-    private static bool ShouldSearchCachedDocuments(
-        [NotNullWhen(true)] out CachedIndexMap? cachedIndexMap,
-        [NotNullWhen(true)] out StringTable? stringTable)
-    {
-        cachedIndexMap = Volatile.Read(ref s_cachedIndexMap);
-        stringTable = Volatile.Read(ref s_stringTable);
-        return cachedIndexMap != null && stringTable != null;
+        // Just drop the map entirely.  The GC will take care of the rest.
+        _cachedIndexMap_DoNotAccessDirectly = null;
```

Comment:
>_cachedIndexMap_DoNotAccessDirectly [](http://example.com/codeflow?start=8&length=35)

Why not just dispose this object here instead of needing it to have a finalizer?